gffquant/profilers/feature_quantifier.py

```
@@ -158,39 +122,53 @@
-            (None, unannotated_reads)[report_unannotated],
-        )
+        total_gene_counts = self.counter.generate_gene_count_matrix(self.reference_manager)
+        logger.info("TOTAL_GENE_COUNTS = %s", total_gene_counts)
 
         count_writer.write_gene_counts(
-            count_annotator.gene_counts,
-            count_annotator.scaling_factors["total_gene_uniq"],
-            count_annotator.scaling_factors["total_gene_ambi"]
+            self.counter,
+            self.reference_manager,
+            u_sf, a_sf,
```

Ruff (0.8.2)

Undefined name `u_sf`

(F821)